api/src/main/java/org/pmiops/workbench/cohortbuilder/SearchGroupItemQueryBuilder.java

```diff
@@ -0,0 +1,578 @@
+        StringBuilder bpSql = new StringBuilder(BP_SQL);
+        List<Long> bpConceptIds = new ArrayList<>();
+        for (Attribute attribute : param.getAttributes()) {
+          if (attribute.getConceptId() != null) {
+            // this indicates we need to build a blood pressure sql statement
+            bpConceptIds.add(attribute.getConceptId());
+            processBloodPressureSql(queryParams, bpSql, attribute);
+          } else if (AttrName.NUM.equals(attribute.getName())) {
+            queryParts.add(processNumericalSql(queryParams, param, attribute));
+          } else {
+            queryParts.add(processCategoricalSql(queryParams, param, attribute));
+          }
+        }
+        if (!bpConceptIds.isEmpty()) {
+          // if blood pressure we need to add named parameters for concept ids
+          QueryParameterValue cids =
+              QueryParameterValue.array(bpConceptIds.stream().toArray(Long[]::new), Long.class);
+          String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+          queryParts.add(String.format(bpSql.toString(), conceptIdsParam) + ")\n");
+        }
+      }
+      // all search parameters in the search group item have the same source/standard flag
+      isStandard = param.getStandard();
+    }
+    // need to OR all query parts together since they exist in the same search group item
+    String queryPartsSql = "(" + String.join(OR, queryParts) + ")\n";
+    if (!childConceptIds.isEmpty()) {
+      // if we have concept ids that are non bp then add named parameters
+      QueryParameterValue cids =
+          QueryParameterValue.array(childConceptIds.stream().toArray(Long[]::new), Long.class);
+      String conceptIdsParam = addQueryParameterValue(queryParams, cids);
+      queryPartsSql = String.format(queryPartsSql, conceptIdsParam);
+    }
+    // create the source/standard flag named parameter
+    String standardParam =
+        addQueryParameterValue(queryParams, QueryParameterValue.int64(isStandard ? 1 : 0));
+    // format the base sql with all query parts
+    String baseSql = String.format(BASE_SQL, standardParam) + queryPartsSql;
+    // build modifier sql if modifiers exists
+    String modifiedSql = buildModifierSql(baseSql, queryParams, searchGroupItem.getModifiers());
+    // split off the sql conditions in case were building a temporal sql statement
+    // otherwise this is just ignored
+    String conditionsSql = baseSql.split("where")[1] + "%s";
+    // build the inner temporal sql if this search group item is temporal
+    // otherwise return modifiedSql
+    return buildInnerTemporalQuery(
+        modifiedSql, conditionsSql, queryParams, searchGroupItem.getModifiers(), mention);
+  }
+
+  /** Build sql statement for demographics */
+  private static String buildDemoSql(
+      Map<String, QueryParameterValue> queryParams, SearchGroupItem searchGroupItem) {
+    // The UI implementation of demographics only allows for 1 search parameter per search group
```

Comment:
Should probably codify anything else as a BadRequest

Reply:
Done.